File: app/common/exceptions/BizException.java
package common.exceptions;

import common.ResponseCode;

public class BizException extends RuntimeException {

    protected int code;
    protected String message;

    public ResponseCode getResponseCode() {
        return responseCode;
    }

    protected ResponseCode responseCode;

    public BizException(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public BizException(String message) {
        this.code = ResponseCode.SYSTEM_ERROR.getCode();
        this.message = message;
    }

    public BizException(ResponseCode responseCode) {
        this.code = responseCode.getCode();
        this.message = responseCode.getMessage();
        this.responseCode = responseCode;
    }
}

File: app/common/exceptions/BizException.scala
package common.exceptions

import common.result.Errors

case class BizException(code: Int, message: String, err: Option[Errors] = None) extends RuntimeException(message) {}

object BizException {

  def apply(err: Errors): BizException = BizException(err.code, err.message, Some(err))

  def apply(errCode: Int, errMsg: String): BizException = {

    val err = Some(
      new Errors {
        override val code: Int       = errCode
        override val message: String = errMsg
      })
    BizException(errCode, errMsg, err)
  }
}
